packages/server/modules/workspaces/domain/operations.ts

```diff
@@ -96,6 +96,12 @@ export type GetWorkspaceRolesForUser = (
 
 export type UpsertWorkspaceRole = (args: WorkspaceAcl) => Promise<void>
 
+export type GetWorkspaceRolesCount = ({
```

Comment:
1. Convert this to getting role counts for multiple workspaces (getWorkspacesRolesCount: () => Promise<Array<{workspaceId, role, count}>>
2. Then you can create a dataloader and use that in the Workspace.cost resolver

Usually converting the query from doing a select for a single entity to a select for a bunch of entities is not very complicated, and gives us an easy performance win in that we have a batched Workspace.cost() resolver that doesn't invoke the query separately for each requested workspace. And if you're ever interested in just 1 workspace, you can still use the query for that.

This should be whenever possible for any queries that are invoked in GQL field resolvers - if you can use a dataloader, do that

Reply:
I considered it, but for now, I decided to skip it for the following reasons:
1. I cannot directly use the dataloader, as we do in other situations, since I need to do some calculations. Considering our agreement, we want those calculations to be in the service, not the resolver or the repository. 
2. this part will be deleted once we will ship a proper pricing solution to the users and the workspace module will be a paying module.
3. We will show this banner in the workspace detail, so a DL wouldn't help